lib/_emerge/actions.py

```diff
@@ -772,6 +772,89 @@ def action_config(settings, trees, myopts, myfiles):
     return retval
 
 
+def get_libc_version(vardb):
+    libcver = []
+    libclist = set()
+    for atom in expand_new_virt(vardb, portage.const.LIBC_PACKAGE_ATOM):
+        if not atom.blocker:
+            libclist.update(vardb.match(atom))
+    if libclist:
+        for cpv in sorted(libclist):
+            libc_split = portage.catpkgsplit(cpv)[1:]
+            if libc_split[-1] == "r0":
+                libc_split = libc_split[:-1]
+            libcver.append("-".join(libc_split))
+        return libcver
+    return ["unavailable"]
+
+
+def get_profilever(portdir, profile, vardb):
+    profilever = None
+    repositories = vardb.settings.repositories
+    if profile:
+        profilever = relative_profile_path(portdir, profile)
+        if profilever is None:
+            try:
+                for parent in portage.grabfile(os.path.join(profile, "parent")):
+                    profilever = relative_profile_path(
+                        portdir, os.path.join(profile, parent)
+                    )
+                    if profilever is not None:
+                        break
+                    colon = parent.find(":")
+                    if colon != -1:
+                        p_repo_name = parent[:colon]
+                        try:
+                            p_repo_loc = repositories.get_location_for_name(p_repo_name)
+                        except KeyError:
+                            pass
+                        else:
+                            profilever = relative_profile_path(
+                                p_repo_loc,
+                                os.path.join(
+                                    p_repo_loc, "profiles", parent[colon + 1 :]
+                                ),
+                            )
+                            if profilever is not None:
+                                break
+            except portage.exception.PortageException:
+                pass
+
+            if profilever is None:
+                try:
+                    profilever = "!" + os.readlink(profile)
+                except OSError:
+                    pass
+
+    if profilever is None:
+        profilever = "unavailable"
+
+    return profilever
+
+
+def emerge_version(emerge_config):
+    portagever = portage.VERSION
+    pythonver = (
+        "python"
+        f" {sys.version_info[0]}"
```

Comment:
Should we do a len check to make sure we never hit index out of bounds?

Reply:
I don't see a need for that, given https://docs.python.org/3/library/sys.html#sys.version_info.